src/main/java/com/google/sps/data/DBInterface.java

```diff
@@ -146,6 +169,49 @@
    */
   public void deleteUserProperty(String userId, String objectId, String collection);
```

Comment:
Are we ever going to use this version of the method? If so, add a comment about thread safety - i.e. what must the user of this method ensure before calling this to avoid a race condition. If not, this should be removed.

Reply:
I think using transactions for non voting operations might be a bit overkill (ie I think its reasonable to assume we won't have 2 or more users logged in to the same user and editing / deleting the same thing at the same time). In which case, they would use the unsafe methods.